### PR TITLE
i3wm: Add debug & logging options

### PR DESCRIPTION
**Pull request by layus**

### CODE REVIEW

File: nixos/modules/services/x11/window-managers/i3.nix [hunk 1]
```diff
@@ -23,6 +23,23 @@ in
...
+      debug = mkOption {
+        default = false;
+        example = true;
+        type = types.bool;
+        description = "Enable debug/verbose logging (see -V option)";
+      };
+
+      logFile = mkOption {
+        default = null;
+        example = "$HOME/.i3/i3log";
```
Comment (jagajaga) [inline nixos/modules/services/x11/window-managers/i3.nix]: Change example to `/var/log/i3.log` please.

Comment (layus) [inline nixos/modules/services/x11/window-managers/i3.nix]: @jagajaga Is this wise ? It would merge the logs of different users...

Comment (jagajaga) [inline nixos/modules/services/x11/window-managers/i3.nix]: Well, then you have to write smth like `${config.users.extraUsers.userName.home}/.i3/i3log`.

Comment (layus) [inline nixos/modules/services/x11/window-managers/i3.nix]: @jagajaga This does not help at all. What if I want different users to have different logfiles ? Shoul I use ${config.users.extraUsers.foo.home}/.i3/i3log or ${config.users.extraUsers.bar.home}/.i3/i3log ? It makes sense for each user to have the logs of its own window manager instance in its own home, no ?

Comment (jagajaga) [inline nixos/modules/services/x11/window-managers/i3.nix]: But of you use $HOME that will be root's home, won't it?

Comment (layus) [inline nixos/modules/services/x11/window-managers/i3.nix]: Not in this context, as the code defined here is used in the xsession script (at https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/x11/display-managers/default.nix#L148). In that script, $HOME is already set to the home of the right user. That script also loads `~/.Xresources` for example.

Comment (jagajaga) [inline nixos/modules/services/x11/window-managers/i3.nix]: Thank you for your explanation. Sorry that I was wrong :)

Comment (layus) [inline nixos/modules/services/x11/window-managers/i3.nix]: Never mind, thank you for taking the time to read me :)

**Comment by devhell** [comment]
Has this been tested? I'm getting errors without adapting my configuration, unless this is the intended behaviour, in which case a warning would probably be better.
The [error](https://headcounter.org/hydra/jobset/hellbuilds/buildit#tabs-errors) in question.

File: nixos/modules/services/x11/window-managers/i3.nix [hunk 2]
```diff
@@ -23,6 +23,23 @@ in
...
+        type = types.string;
```
Comment (jagajaga) [inline nixos/modules/services/x11/window-managers/i3.nix]: You have to define that it can be null or string.

Comment (devhell) [inline nixos/modules/services/x11/window-managers/i3.nix]: Can we not have a sane default??

Comment (layus) [inline nixos/modules/services/x11/window-managers/i3.nix]: The previous behaviour was to put the logs in display-manager.unit logs, that is, writing everything to stdout within the xsession script. This is what I wanted to preserve with the default null value.
Tested then thought smart to add the type option... Apologies for the annoyance.

Comment (devhell) [inline nixos/modules/services/x11/window-managers/i3.nix]: That's fine, except that if the user is not aware of this and doesn't configure it (as was the case with my automated build) then that evaluation error comes up.

Comment (devhell) [inline nixos/modules/services/x11/window-managers/i3.nix]: Other than that I like your idea. ;-)

**Pull request merged**